src/pool.rs

```diff
@@ -375,8 +375,8 @@ impl Pool {
                                          }
                                      };
                                      let old =
-                                         shared.finished.fetch_add(1, atomic::Ordering::SeqCst);
-                                     let sent = shared.sent.load(atomic::Ordering::SeqCst);
+                                         shared.finished.fetch_add(1, atomic::Ordering::AcqRel);
```

Comment:
I'm interested to hear your motivation for these orderings. It seems to me that they could even be `Relaxed`, since (a) the `send`/`recv` (essentially) gives a Acquire/Release connection between the `fetch_add` of `sent` below, and the load of `sent`, and (b) these are a "optimisation"/"hint", and aren't critical to correctness.